src/subscription_manager/cache.py

```diff
@@ -1025,8 +1025,22 @@ def _is_cache_obsoleted(self, uep, identity, *args, **kwargs):
         if uep is None:
             cp_provider = inj.require(inj.CP_PROVIDER)
             uep = cp_provider.get_consumer_auth_cp()
-        if hasattr(uep.conn, 'is_consumer_cert_key_valid') and uep.conn.is_consumer_cert_key_valid is True:
-            return False
+
+        if hasattr(uep.conn, 'is_consumer_cert_key_valid'):
+            if uep.conn.is_consumer_cert_key_valid is None:
+                log.debug(
+                    f'Cache file {self.CACHE_FILE} cannot be considered as valid, because no connection has not '
```

Comment:
I think this sentence is more accurate without the "not"

Reply:
Thanks! Created #2914 for this, will append to the commits of this PR once approved.